addons/protected_fields/models.py

```diff
@@ -52,12 +52,13 @@ def _check_field_permissions(self, vals, permitted_vals=None):
     def create(self, vals):
         # default values are permitted upon creation
         defaults = self.default_get(vals)
-        self._check_field_permissions(vals, permitted_vals=defaults)
         record = super(ProtectedFieldsMixin, self).create(vals)
+        record._check_field_permissions(vals, permitted_vals=defaults)
         return record
 
     @api.multi
     def write(self, vals):
         self._check_field_permissions(vals)
-        record = super(ProtectedFieldsMixin, self).write(vals)
-        return record
+        success = super(ProtectedFieldsMixin, self).write(vals)
+        self._check_field_permissions(vals)
```

Comment:
But if you feel uncomfortable with this solution I can do all this ("save" and check) in draft mode (changes made in draft mode are never send to the db) and only then performed another save - this time outside of the draft mode. Would it be better?